Kernel/Bus/USB/xHCI/xHCIController.cpp

```diff
@@ -0,0 +1,1441 @@
+        return EINVAL;
+    auto* extended_capabilities_pointer = m_registers_mapping.ptr() + (m_capability_registers.capability_parameters_1.xHCI_extended_capabilities_pointer << 2);
+
+    ExtendedCapability volatile* extended_capability = nullptr;
+    Vector<SupportedProtocolExtendedCapability volatile*> supported_protocol_capabilities;
+    while (true) {
+        extended_capability = reinterpret_cast<ExtendedCapability volatile*>(extended_capabilities_pointer);
+        if (extended_capability->capability_id == ExtendedCapability::CapabilityID::Supported_Protocols)
+            TRY(supported_protocol_capabilities.try_append(reinterpret_cast<SupportedProtocolExtendedCapability volatile*>(extended_capability)));
+        u32 const next_extended_capability_offset = extended_capability->next_xHCI_extended_capability_pointer << 2;
+        if (next_extended_capability_offset == 0)
+            break;
+        extended_capabilities_pointer += next_extended_capability_offset;
+    };
+
+    if (supported_protocol_capabilities.is_empty())
+        return EINVAL;
+
+    for (auto* supported_protocol_capability : supported_protocol_capabilities) {
+        if (supported_protocol_capability->name_string != SupportedProtocolExtendedCapability::usb_name_string)
+            continue;
+        auto major_revision = supported_protocol_capability->major_revision;
+        if (major_revision < 0x02 || major_revision > 0x03)
+            continue;
+        auto minor_revision = supported_protocol_capability->minor_revision;
+        if (major_revision == 0x02 && minor_revision != 0x00)
+            continue;
+        if (major_revision == 0x03 && minor_revision > 0x20)
+            continue;
+        auto offset = supported_protocol_capability->compatible_port_offset;
+        if (offset < 1 || offset > m_port_max_speeds.size())
+            continue;
+        auto count = supported_protocol_capability->compatible_port_count;
+        if (count == 0 || count > (m_port_max_speeds.size() - offset + 1))
+            continue;
+        if (supported_protocol_capability->protocol_speed_id_count > 0) {
+            dmesgln_pci(*this, "Controller has explicit protocol speed ID definitions - this is not supported yet");
+            continue;
+        }
+        auto max_speed = major_revision == 0x03 ? USB::Device::DeviceSpeed::SuperSpeed : USB::Device::DeviceSpeed::HighSpeed;
+        for (auto i = offset - 1; i < count; ++i)
+            m_port_max_speeds[i] = max_speed;
+    }
+    return {};
+}
+
+ErrorOr<void> xHCIController::initialize()
+{
+    dmesgln_pci(*this, "Controller found {}", PCI::get_hardware_id(device_identifier()));
+    dmesgln_pci(*this, "Registers base: {}", m_registers_mapping.paddr);
+
+    auto interface_version = m_capability_registers.host_controller_interface_version_number;
+    if (interface_version < 0x0090 || interface_version > 0x0102) { // The Intel specification defines versions 0.9.0 up to 1.2.0
```

Comment:
The `0x0102` should be a `0x0120` (1.2.0 instead of 1.0.2).

With that change this driver also works on my main pc!